test-result-summary-client/src/utils/parseJenkinsUrl.js

```diff
@@ -0,0 +1,29 @@
+export const parseJenkinsUrl = (jenkinsUrl, compareType="") => {
+    let parsedRes = { errorMsg: "",serverUrl: "", buildName: "", buildNum: ""};
+    if (!jenkinsUrl) {
+        parsedRes.errorMsg = "Please provide " + compareType + " Build URL."
+        return parsedRes;
+    }
+    let parsedParams = jenkinsUrl.split("/");
+
+    // Find the index for the top level "job" path in the Jenkins URLs given.
+    // This is to support comparing the following equivalent Jenkins job URLs:
+    // https://customJenkinsServer/view/PerfTests/job/Daily-Liberty-DayTrader3/155/
+    // https://customJenkinsServer/job/Daily-Liberty-DayTrader3/155/
+    try {
+        parsedRes.serverUrl = parsedParams[0] + "//" + parsedParams[2];
+        let regNum = /^\d+$/;
+        parsedRes.buildNum = parsedParams.pop();
+        // handle possible trailing "/" in url
+        if(!regNum.test(parsedRes.buildNum)) {
+            parsedRes.buildNum = parsedParams.pop();
+            if(!regNum.test(parsedRes.buildNum)) {
+                throw new Error("error");
```

Comment:
Please be more specific about the error msg. i.e., cannot find build number

Reply:
Error message has been added detailed info, and integrated into errorMsg, an example is as following:
![image](https://user-images.githubusercontent.com/48260423/82587988-1e91d800-9b68-11ea-95af-e1684e01b07e.png)